crates/uv-settings/src/settings.rs

```diff
@@ -484,29 +805,287 @@ pub struct PipOptions {
+            emit-binary = true
+        "#
+    )]
     pub emit_build_options: Option<bool>,
+    /// Whether to emit a marker string indicating the conditions under which the set of pinned
+    /// dependencies is valid.
+    ///
+    /// The pinned dependencies may be valid even when the marker expression is
+    /// false, but when the expression is true, the requirements are known to
+    /// be correct.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            emit-marker-expression = true
+        "#
+    )]
     pub emit_marker_expression: Option<bool>,
+    /// Include comment annotations indicating the index used to resolve each package (e.g.,
+    /// `# from https://pypi.org/simple`).
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            emit-index-annotation = true
+        "#
+    )]
     pub emit_index_annotation: Option<bool>,
+    /// The style of the annotation comments included in the output file, used to indicate the
+    /// source of each package.
+    #[option(
+        default = "\"split\"",
+        value_type = "str",
+        example = r#"
+            annotation-style = "line"
+        "#
+    )]
     pub annotation_style: Option<AnnotationStyle>,
+    /// The method to use when installing packages from the global cache.
+    ///
+    /// Defaults to `clone` (also known as Copy-on-Write) on macOS, and `hardlink` on Linux and
+    /// Windows.
+    #[option(
+        default = "\"clone\" (macOS) or \"hardlink\" (Linux, Windows)",
+        value_type = "str",
+        example = r#"
+            link-mode = "copy"
+        "#
+    )]
     pub link_mode: Option<LinkMode>,
+    /// Compile Python files to bytecode after installation.
+    ///
+    /// By default, does not compile Python (`.py`) files to bytecode (`__pycache__/*.pyc`), instead
+    /// Python lazily does the compilation the first time a module is imported. In cases where the
+    /// first start time matters, such as CLI applications and docker containers, this option can
+    /// trade longer install time for faster startup.
+    ///
+    /// The compile option will process the entire site-packages directory for consistency and
+    /// (like pip) ignore all errors.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            compile-bytecode = true
+        "#
+    )]
     pub compile_bytecode: Option<bool>,
+    /// Require a matching hash for each requirement.
+    ///
+    /// Hash-checking mode is all or nothing. If enabled, _all_ requirements must be provided
+    /// with a corresponding hash or set of hashes. Additionally, if enabled, _all_ requirements
+    /// must either be pinned to exact versions (e.g., `==1.0.0`), or be specified via direct URL.
+    ///
+    /// Hash-checking mode introduces a number of additional constraints:
+    /// - Git dependencies are not supported.
```

Comment:
Some markdown renderers are picky about a newline here.

Reply:
Ah yeah good call. I think MkDocs is picky about that, actually.